src/main.rs

```diff
@@ -1,26 +1,125 @@
-use structopt::StructOpt;
+mod config;
 use colored::Colorize;
+use structopt::StructOpt;
+
+extern crate rand;
+use rand::thread_rng;
+use rand::Rng;
 
+use fake::faker::company::en::*;
+use fake::{Fake, Faker};
+
+use std::{thread, time};
+
+mod dsm;
 #[derive(Debug, StructOpt)]
 #[structopt()]
 pub struct Opt {
-    /// Limit scan to given applications
-    #[structopt(short, long = "app-slug", name = "slug", about = "App name that will receive injections")]
+    /// Limit scan to given applications. App name(s) that will receive injections.
+    #[structopt(short, long = "app-slug", name = "slug")]
     app_slugs: Vec<String>,
-    /// Maximum count of forms to process
-    #[structopt(short = "l", long, name = "count", about = "Number data source that will be injected")]
-    dsm_limit: Option<usize>,
+    /// Number data source that will be injected. Number needs to be bigger than 1.
+    #[structopt(short = "l", long, name = "limit")]
+    dsm_limit: Option<i32>,
+}
+
+async fn process(opt: &Opt, token: String) {
+    // Set limit for dsm
+    let limit = match opt.dsm_limit {
+        Some(val) => {
+            if val > 1 {
+                opt.dsm_limit.unwrap()
+            } else {
+                println!("{}", "Invalid dsm limit. Will default to 10".yellow());
+                10
+            }
+        }
+        _ => 10,
+    };
```

Comment:
An alternative way of writing this would look something like the following:

```suggestion
    let limit = match opt.dsm_limit {
        Some(val) if val > 1 => opt.dsm_limit.unwrap(),
        Some(val) => {
            println!("{}", "Invalid dsm limit. Will default to 10".yellow());
            10
        }
        None => 10,
    };
```

Sometimes it can look a little cleaner, but just sharing some optional alternative syntax.